specification/compiler/steps/validate-model.ts

```diff
@@ -0,0 +1,594 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as model from '../model/metamodel'
+
+enum TypeDefKind {
+  type,
+  behavior
+}
+
+/**
+ * Validates the internal consistency of the model (doesn't check the json spec)
+ *
+ * Any inconsistency is logged as an error.
+ *
+ * Missing validations:
+ * - verify uniqueness of property names in the inheritance chain
+ * - verify that request parents don't define properties (would they be path/request/body properties?)
+ */
+export default function validateModel (apiModel: model.Model, failHard: boolean): model.Model {
```

Comment:
As said above I would keep the export function as a step declaration, and read the `failHard` from env or cli args.

Reply:
Done. It now checks if the `FAIL_HARD` env variable is set.